source/client/ui/explorer/LanguageMenu.ts

```diff
@@ -84,7 +84,7 @@ export default class LanguageMenu extends Popup
             </div>
             <div class="ff-flex-row">
                 <div class="ff-scroll-y sv-scroll-offset" role="listbox">
-                    ${language.activeLanguages.map((language, index) => this.renderEntry(language, index))}
+                    <sv-property-options class="ff-flex-column" buttons .property=${language.ins.language}></sv-property-options>
```

Comment:
I may be missing something from another commit, but this doesn't seem correct to me. language.ins.language should be all supported languages (i.e. all languages we have dictionaries for) and the Explorer language menu should have all languages actively used in the scene.

Reply:
I opted to split the "list of accepted languages" (a subset of "all dictionaries") into `language.ins.languages`, whose value is then used as possible options for `language.ins.language` ; then always propose just this subset of languages until another language is enabled in the scene.

The reason behind this is something I saw a lot of users do during scene editing :
- edit some annotation/article in a french-only scene
- click on "Français", switch to "English" (misclick, or looking for something else, or not knowing this scene is not translated)
- Save the scene at any point in the future
- The scene is now FR/EN forever, even though most/all EN values are undefined. There is no way to go back to "french-only"

To prevent this I rather liked the concept of this intermediate list of languages that makes it unequivocaly clear what languages are expected to be supported in this scene.

We might go the other route and not have this list. The list of available languages would then be "any language for which something is defined" but then : 

1. we can't easily delete a language for a scene once it has been touched
2. we have to thoroughly search through the scene for all translation dictionaries
3. inside the Story editor we will have all available languages to select from. Keeping in mind that I might be required to commit to some basic localization for a large chunk of languages spoken in the EU in the near future, that's a quite a list to have.

Reply:
Thanks for the explanation. I understand the motivation, and acknowledge the issue, but am not sure I agree this is the best route forward.

> * edit some annotation/article in a french-only scene
> * click on "Français", switch to "English" (misclick, or looking for something else, or not knowing this scene is not translated)
> * Save the scene at any point in the future
> * The scene is now FR/EN forever, even though most/all EN values are undefined. There is no way to go back to "french-only"

I think this is only true with articles, as switching languages currently creates a new html article asset. With annotations and titles, if the data is undefined on save it shouldn't be getting written to the file. If there are cases where it is, that sound like a bug. 

> 
> To prevent this I rather liked the concept of this intermediate list of languages that makes it unequivocaly clear what languages are expected to be supported in this scene.
> 

How do you see this list being defined? From my current perspective, this isn't something we could expect from our users.

> We might go the other route and not have this list. The list of available languages would then be "any language for which something is defined" but then :
> 
> 1. we can't easily delete a language for a scene once it has been touched
> 2. we have to thoroughly search through the scene for all translation dictionaries
> 3. inside the Story editor we will have all available languages to select from. Keeping in mind that I might be required to commit to some basic localization for a large chunk of languages spoken in the EU in the near future, that's a quite a list to have.

For 3. this is intentional. I don't think it's any different from standard localization lists in many sites/apps.

I'm going to think a little bit more on alternatives to address this issue.